tests/PS09.md

```diff
@@ -0,0 +1,595 @@
+
+
+###### Step 5:
+
+Server replies with *CanStoreDataResponse*:
+  
+  * `Message.id == 6`
+  * `Response.status == STATUS_OK`
+
+Server replies with *CanPublishIpnsRecordResponse*:
+  
+  * `Message.id == 7`
+  * `Response.status == STATUS_OK`
+
+
+###### Step 6:
+
+CAN server successfully resolves the test identity IPNS to IPFS path, which matches $ObjectPath2.
+
+CAN server successfully provides CAN object that matches $IdentityData2.
+
+
+###### Step 7:
+
+Server replies with *CanStoreDataResponse*:
+  
+  * `Message.id == 8`
+  * `Response.status == STATUS_OK`
+
+CAN server reports that there are no pints for $ObjectPath2.
+
+
+###### Step 8:
+
+CAN server reports that there are no pints for $ObjectPath1.
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+#### PS09003 - CAN Store Data - Invalid Hosting Server ID
+
+##### Prerequisites/Inputs
+  * Server's database is empty.
+  * CAN server is running on localhost
+
+###### Inputs:
+  * Server's IP address
+  * Server's primary port
+
+##### Description 
+
+The test tries to store object in CAN without setting the right hosting server ID.
+
+
+###### Step 1:
+
+The tests obtains a list of service ports from the profile server's primary port.
+
+The test establishes a hosting agreement with the profile server for its identity and it checks-it in.
+
+
+###### Step 2:
+
+The test sends *CanStoreDataRequest*:
+
+  * `Message.id := 3`
+  * `CanStoreDataRequest.data.hostingServerId` is set to 5 bytes 0x40.
+  * `CanStoreDataRequest.data.keyValueList := 
+     [
+       { key := "key1", stringValue := "value 1" },
+       { key := "key2", uint32Value := 2 },
+       { key := "key3", boolValue := true },
+       { key := "key4", binaryValue := [1,2,3] },
+     ]`
+ 
+and reads the response. 
+
+ 
+
+##### Acceptance Criteria
+
+###### Step 1:
+
+The test successfully obtains list of ports on which the profile server provides its services. 
+Then the test successfully establishes a hosting agreement for its identity and checks-it in.
+
+
+###### Step 2:
+
+Server replies with *Response*:
+  
+  * `Message.id == 3`
+  * `Response.status == ERROR_INVALID_VALUE`
+  * `Response.details == "data.hostingServerId"`
+
+
+
+
+
+
+
+
+#### PS09004 - CAN IPNS Record - Invalid Requests
+
+##### Prerequisites/Inputs
+  * Server's database is empty.
+  * CAN server is running on localhost
+
+###### Inputs:
+  * Server's IP address
+  * Server's primary port
+
+##### Description 
+
+The test sends a series of invalid requests related to publishing IPNS record to test error handling on the server side.
+
+
+###### Step 1:
+
+The tests obtains a list of service ports from the profile server's primary port.
+
+The test establishes a hosting agreement with the profile server for its identity and it checks-it in.
+
+
+###### Step 2:
+
+The test sends *CanPublishIpnsRecordRequest*:
+
+  * `Message.id := 3`
+  * `CanPublishIpnsRecordRequest.record.value := "/ipfs/QmaRvpLT4RgBDVY6cQKwqQGQNtbM31KSqh8TxvQgFoEUus"`

```

Comment:
Is it some hash value that cannot be parsed by IPFS or is it something you just made up, so it is unlikely to exist on the network?